cmake/std/trilinosprhelpers/setenvironment/unittests/test_SetEnvironment.py

```diff
@@ -583,7 +583,110 @@ def test_SetEnvironment_module_returns_NoneType(self):
                 setEnv.apply()
 
 
-    def _setEnv_test(self, filename, profile, truth=None, module_fail=False):
+    def test_SetEnvironment_cyclebreaking_01(self):
+        """
+        Test that the cycle-breaking logic works properly.
+        - Load [Sec_D] from the test config.ini file.
+        - The `use Sec_C:` entry from Sec_A should be blocked from running.
+        - The envvar Sec_C should be set to '99'
+        """
+        setEnv = SetEnvironment(self._filename, "Sec_D")
+
+        with patch('trilinosprhelpers.setenvironment.ModuleHelper.module', side_effect=mock_module_pass):
+            setEnv.apply()
+
+        assert "SEC_A" in os.environ.keys()
+        assert "SEC_B" in os.environ.keys()
+        assert "SEC_C" in os.environ.keys()
+        assert "SEC_D" in os.environ.keys()
+        print("SEC_A == {}".format(os.environ["SEC_A"]))
+        print("SEC_B == {}".format(os.environ["SEC_B"]))
+        print("SEC_C == {}".format(os.environ["SEC_C"]))
+        print("SEC_D == {}".format(os.environ["SEC_D"]))
+        assert os.environ["SEC_A"] == '1'
+        assert os.environ["SEC_B"] == '1'
+        assert os.environ["SEC_C"] == '99'
+        assert os.environ["SEC_D"] == '1'
+
+
+    def test_SetEnvironment_cyclebreaking_02(self):
+        """
+        Test that the cycle-breaking logic works properly.
+        - Load [Sec_D2] from the test config.ini file.
+        - The second call to `use Sec_A2` in `Sec_D2` should work properly.
+        - The envvar `Sec_A2` should end up being set to 1.
+        """
+        setEnv = SetEnvironment(self._filename, "Sec_D2")
+
+        with patch('trilinosprhelpers.setenvironment.ModuleHelper.module', side_effect=mock_module_pass):
+            setEnv.apply()
+
+        assert "SEC_A2" in os.environ.keys()
+        assert "SEC_B2" in os.environ.keys()
+        assert "SEC_C2" in os.environ.keys()
+        assert "SEC_D2" in os.environ.keys()
+        print("SEC_A2 == {}".format(os.environ["SEC_A2"]))
+        print("SEC_B2 == {}".format(os.environ["SEC_B2"]))
+        print("SEC_C2 == {}".format(os.environ["SEC_C2"]))
+        print("SEC_D2 == {}".format(os.environ["SEC_D2"]))
+        assert os.environ["SEC_A2"] == '1'
+        assert os.environ["SEC_B2"] == '1'
+        assert os.environ["SEC_C2"] == '1'
+        assert os.environ["SEC_D2"] == '1'
+
+
+    def test_SetEnvironment_TEST_MODULE_FAIL(self):
+        """
+        Test loading TEST_MODULE_FAIL
+        - Tests loading a nonexistent module
+        - Module load should fail but we don't throw an exception.
+        - apply() should return a nonzero value.
+        """
+        filename    = self._filename
+        profile     = "TEST_MODULE_FAIL"
+        module_fail = True
+        throw_on_fail = False
```

Comment:
Do you want the `=`s aligned?  PEP8 says "no", but I know how you love your whitespace 😁

Reply:
PEP8 can be stupid sometimes.

Reply:
Oh I agree.  I was also trying to point out that the `=` signs aren’t all vertically aligned, which I thought is what you wanted.